src/pages/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+
+import { Loader } from '../../components/Loader';
+import { PeopleTable } from '../../components/PeopleTable';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    getPeople()
+      .then(response => setPeople(response))
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoaded(true));
+  }, []);
+
+  const preparedPeople: Person[] = people.map(person => {
```

Comment:
The right decision, but now you have two searches in the middle of another search, which is algorithmically not optimal.
1. To improve the algorithm, you can create a separate map (`peopleMap`) and add all the people to it using their name as a key (`peopleMap.set(person.name, person)`), after that go through the `people` array and find the person's father and mother (`peopleMap.get(person.motherName)`, `peopleMap.get(person.fatherName)`).
This is a more complicated approach, but more correct on real projects.
2. For code cleanliness, this function could be moved outside the component and used as a helper.
3. Now, with each rendering of the component, you will search and prepare the `preparedPeople` array - this can be improved by calling this search in the `useEffect` hook.